### ISSUE TITLE
Enable directional derivative calculation by generating complete Jacobian and multiply

### ISSUE THREAD

**Issue by jaeandersson** [casadi/casadi#1273]
This has been implemented (see function_internal.cpp), but is not enabled. Should probably be done after #700 and needs to be tested carefully to avoid adverse effects.

**Comment by jaeandersson** [comment]
Postponed to 2.3.

**Comment by jaeandersson** [comment]
Now enabled by default.
